pkg/controller/sync/accessor.go

```diff
@@ -0,0 +1,275 @@
+	finalizerClient := templateClient
+	if a.targetIsNamespace {
+		// For namespaces the finalizer is applied to the placement to
+		// ensure that deletion of namespaces in member clusters only
+		// occurs after a namespace has been configured to be propagated
+		// to those clusters.
+		finalizerClient = placementClient
+	}
+
+	a.deletionHelper = deletionhelper.NewDeletionHelper(
+		func(rawObj pkgruntime.Object) (pkgruntime.Object, error) {
+			obj := rawObj.(*unstructured.Unstructured)
+			return finalizerClient.Resources(obj.GetNamespace()).Update(obj, metav1.UpdateOptions{})
+		},
+		func(obj pkgruntime.Object) string {
+			return util.NewQualifiedName(obj).String()
+		},
+		informer,
+		updater,
+	)
+
+	return a, nil
+}
+
+func (a *resourceAccessor) Run(stopChan <-chan struct{}) {
+	go a.versionManager.Sync(stopChan)
+	go a.templateController.Run(stopChan)
+	go a.overrideController.Run(stopChan)
+	go a.placementPlugin.Run(stopChan)
+}
+
+func (a *resourceAccessor) HasSynced() bool {
+	if !a.versionManager.HasSynced() {
+		glog.V(2).Infof("Version manager not synced")
+		return false
+	}
+	if !a.templateController.HasSynced() {
+		glog.V(2).Infof("Templates not synced")
+		return false
+	}
+	if !a.overrideController.HasSynced() {
+		glog.V(2).Infof("Overrides not synced")
+		return false
+	}
+	if !a.placementPlugin.HasSynced() {
+		glog.V(2).Infof("Placements not synced")
+		return false
+	}
+	return true
+}
+
+func (a *resourceAccessor) FederatedResource(eventSource util.QualifiedName) (FederatedResource, error) {
+	if a.targetIsNamespace && a.isSystemNamespace(eventSource.Name) {
+		glog.V(7).Infof("Ignoring system namespace %q", eventSource.Name)
+		return nil, nil
+	}
+
+	// Most federated resources have the same name as their targets.
+	targetName := eventSource
+	federatedName := util.QualifiedName{
+		Namespace: eventSource.Namespace,
+		Name:      eventSource.Name,
+	}
+	templateKey := federatedName.String()
+
+	// If the target type is namespace, the placement resource must be
+	// present and will be used as the finalization target.
+	var placement *unstructured.Unstructured
+
+	// A federated primitive for namespace "foo" is namespaced
+	// (e.g. "foo/foo"). An event sourced from a namespace in the host
+	// or member clusters will have the name "foo", and an event
+	// sourced from a federated resource will have the name "foo/foo".
+	// In order to ensure object retrieval from the informers, it is
+	// necessary to derive the target name and federated name from the
+	// event source.
+	if a.targetIsNamespace {
+		eventSourceIsTarget := eventSource.Namespace == ""
+		if eventSourceIsTarget {
+			// Ensure the federated name is namespace qualified.
+			federatedName.Namespace = federatedName.Name
+		} else {
+			// Ensure the target name is not namespace qualified.
+			targetName.Namespace = ""
+		}
+
+		// A namespace is only federated if it has a corresponding
+		// placement resource.
+		var err error
```

Comment:
:= seems more concise for err definition

Reply:
It's not possible to use `:=` when one of the targets is already defined (in this case, as a struct variable.